src/directory_listing/mod.rs

```diff
@@ -128,7 +123,7 @@ impl DirectoryListing {
         // TODO try using the below approach for efficiency - also try the same in upsert_sub_directory
         // if let Some(mut existing_file) = self.files.iter_mut().find(|entry| *entry.get_name() == *file.get_name()) {
         // *existing_file = file;
-        if let Some(index) = self.files.iter().position(|entry| *entry.get_name() == *file.get_name()) {
+        if let Some(index) = self.files.iter().position(|entry| *entry.get_id() == *file.get_id()) {
```

Comment:
Need to make sure that: `if name_lhs == name_rhs` then `id_lhs == id_rhs` else reject saying `NfsError::FileNameConflict`